portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -23,10 +25,21 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
-
-    Key commentEntityKey = KeyFactory.createKey("Comment", id);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(commentEntityKey);
+    UserService userService = UserServiceFactory.getUserService();
+    String commentEmail = request.getParameter("email");
+    String userEmail = request.getParameter("user");
+    //if user did not make the comment or if user is not logged in 
+    if (!commentEmail.equals(userEmail) || !userService.isUserLoggedIn()){
+      System.err.println("ERROR: Cannot delete comment.");
+      return;
+    }
+    try {
+      long id = Long.parseLong(request.getParameter("id"));
+      Key commentEntityKey = KeyFactory.createKey("Comment", id);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.delete(commentEntityKey);
```

Comment:
When dealing with `try`/`catch`, try to keep as little logic as possible within the try statement. In this case I could recommend one of two fixes:

- Implement a `getCommentId(WHATEVER_PARAMS...)` which handles the try catch and returns a long with -1 or 0 when its invalid
- OR if you still want this try catch in this method:
```java
long id;
try {
  id = Long.parseLong(request.getParameter("id"));
} catch (NumberFormatException e){
  System.err.println("ERROR: Failed to parse to long.");
  // Invalid id, don't continue
  return;
}

...
```